### PR TITLE
add timer_task, proposal and govern_token

### PR DESCRIPTION
**Pull request by Tri-stone**
(1) add timer tasks
(2) add proposal
(3) add govern token
(4)add authority control for specific kernel contracts

### CODE REVIEW

File: bcs/ledger/xledger/state/tx_verification.go [hunk 1]
```diff
@@ -585,6 +641,25 @@ func (t *State) verifyTxRWSets(tx *pb.Transaction) (bool, error) {
 	return true, nil
 }
 
+// verifyAutoTxRWSets verify auto tx read sets and write sets
+func (t *State) verifyAutoTxRWSets(tx, autoTx *pb.Transaction) (bool, error) {
```
Comment (chunhui01) [inline bcs/ledger/xledger/state/tx_verification.go:645]: This method doesn't work.

**Review by chunhui01 — APPROVED** [review]

File: bcs/ledger/xledger/state/tx_verification.go (hunk 1 above)
Comment (Tri-stone) [reply bcs/ledger/xledger/state/tx_verification.go:646]: done

**Review by ownluke — APPROVED** [review]

File: kernel/contract/proposal/timer/timer_task_contract.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	blockHeightBuf := args["block_height"]
+	if blockHeightBuf == nil {
+		return nil, fmt.Errorf("do timer tasks failed, blockHeightBuf is nil")
+	}
+	// 根据高度遍历所有提案，判断是否达到投票要求并执行
+	startKey := utils.MakeTimerBlockHeightPrefix(string(blockHeightBuf))
+	prefix := utils.MakeTimerBlockHeightPrefixSeparator(string(blockHeightBuf))
+	endKey := utils.PrefixRange([]byte(prefix))
+	iter, err := ctx.Select(utils.GetTimerBucket(), []byte(startKey), endKey)
+	defer iter.Close()
```
Comment (icexin) [inline kernel/contract/proposal/timer/timer_task_contract.go:60]: 先检查错误，再Close

File: bcs/ledger/xledger/ledger/genesis.go [hunk 3]
```diff
@@ -148,6 +153,26 @@ func (rc *RootConfig) GetReservedWhitelistAccount() string {
 	return rc.ReservedWhitelist.Account
 }
 
+// GetPredistribution return predistribution
+func (rc *RootConfig) GetPredistribution() []Predistribution {
+	return PredistributionTranslator(rc.Predistribution)
+}
+
+func PredistributionTranslator(predistribution []struct {
```
[the remark this answers is not in the document]
Comment (Tri-stone) [reply bcs/ledger/xledger/ledger/genesis.go:161]: genesis里用了临时结构体。
golang强类型，用自己定义的，类型校验不过。

File: kernel/contract/proposal/timer/timer_task_contract.go (hunk 2 above)
Comment (Tri-stone) [reply kernel/contract/proposal/timer/timer_task_contract.go]: done

**Pull request merged**